app/javascript/src/components/Player.tsx

```diff
@@ -139,18 +141,63 @@ class Player extends React.Component<PlayerProps, PlayerState> {
     )
   }
 
-  // If there is no modifier always return true
-  // If there is a modifier make sure that the user has it
-  private showIf(port: DefaultPortModel): boolean {
-    const modifier: string | undefined = get(this.props.portMeta as any, `${port.id}.showIf`)
-    return !modifier || this.hasModifier(modifier)
+  private showIfItem(port: DefaultPortModel): boolean {
+    const { currentItems } = this.state
+    const showIfItems: ShowIfItem[] = get(this.props.portMeta as any, `${port.id}.showIfItems`)
+
+    // If no data set, show the option
+    if (!showIfItems) return true
+
+    // Otherwise, loop over showIf config
+    // Bail if a negative case is found, otherwise return true
+    for (let showIf of showIfItems) {
+      // Should have it but doesn't
+      if (showIf.hasIt && currentItems.indexOf(showIf.name) === -1) {
+        return false
+      }
+
+      // Should not have it but does
+      if (!showIf.hasIt && currentItems.indexOf(showIf.name) !== -1) {
+        return false
+      }
+    }
+
+    return true
   }
 
-  // If there is no modifier always return true
-  // If there is a modifier make sure that the user does not have it
-  private showUnless(port: DefaultPortModel): boolean {
-    const modifier: string | undefined = get(this.props.portMeta as any, `${port.id}.showUnless`)
-    return !modifier || !this.hasModifier(modifier)
+  private showIfStat(port: DefaultPortModel): boolean {
+    const { currentItems } = this.state
+    const showIfStats: ShowIfStat[] = get(this.props.portMeta as any, `${port.id}.showIfStats`)
+    // const playerStats: PlayerStat[] = get(this.props.portMeta as any, `${port.id}.statChanges`)
+
+    // this.props.portMeta.some(test)
+    // If no data set, show the option
+    if (!showIfStats) return true
+
+    // Otherwise, loop over showIf config
+    for (let showIf of showIfStats) {
+      // Get the stat related to this stat check
+      let currentStat = (Object.values(this.props.portMeta).find(obj => {
+        return obj.statChanges![0].name === showIf.name
+      }))!.statChanges![0]
```

Comment:
@joeyjoejoejr @DanielSBrown @solomonhawk 

This new modifier will track a player's stats through `statChanges`, and will have a `showIf` down the line for that specific stat to be greater than, less than, etc. of a certain value (e.g. a Player drinks a potion at step 2 that sets their health to 10. In step 3, only show an choice if `Health >= 10`. 

This bit of code finding the specific stat by its name when comparing to a `showIf`'s name. So if the player had two stats, `Health` and `Damage`, but the `showIf` I'm currently evaluating is `Health`, only grab that stat and its value. I feel like there has to be a better way to do this, and wanted to get feedback on how to achieve that. The multiple !s and indexing of the array at 0 "smells" really bad to me. 

I see now that I might have to aggregate all of the changes made to a specific stat up to that point, but I wanted to get feedback on how this could be bettered to make it less clunky looking if possible.

Reply:
I think this is better to talk out over a chat.  I'm happy to talk you through this.  I think I see what the cause of this is.

Reply:
FWIW: I think this should work similarly to `currentItems`, which is added to / removed from when you make choices: [code](https://github.com/vigetlabs/storyboard/blob/332dccfc941d711d67b6083016e9d6ac7a8b19a3/app/javascript/src/components/Player.tsx#L255-L270). Tracking stats in a simple struct on `this.state` that can be easily referenced here.